File: LeetCode/Study Plan/Practice Problems/Easy/Two Sum/Go/two_sum.go
package main

import "fmt"

func two_sum(arr []int, target int) []int {
	mp := make(map[int]int)
	for i := 0; i < len(arr); i++ {
		mp[arr[i]] = i
	}
	res := []int{}
	for i := 0; i < len(arr); i++ {
		val := target - arr[i]
		if _, ok := mp[val]; ok {
			if i != mp[val] {
				res = append(res, []int{mp[val], i}...)
				break
			}
		}
	}
	return res
}

func main() {
	fmt.Println(two_sum([]int{2, 7, 11, 15}, 9))
}

File: LeetCode/Study Plan/Practice Problems/Easy/Two Sum/Java/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numInd = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            int rem = target-nums[i];
            if (numInd.containsKey(rem)) return new int[]{i, numInd.get(rem)};
            numInd.put(nums[i], i);
        }
        return null;
    }
}
